core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -0,0 +1,397 @@
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = HttpData.of(contentType.charset(StandardCharsets.UTF_8), content);
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType contentType, String content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = HttpData.of(contentType.charset(StandardCharsets.UTF_8), content);
+        return this;
+    }
+
+    /**
+     * Sets the content for this request. The content of the request is formatted by
+     * {@link String#format(Locale, String, Object...)} with {@linkplain Locale#ENGLISH English locale}.
+     */
+    @FormatMethod
+    public HttpRequestBuilder content(MediaType contentType, @FormatString String format, Object... content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(format, "format");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = HttpData.of(contentType.charset(StandardCharsets.UTF_8), format, content);
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType contentType, byte[] content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = HttpData.wrap(content);
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType contentType, HttpData content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = content;
+        return this;
+    }
+
+    /**
+     * Sets a header for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .header("authorization", "foo")
+     *            .build();
+     * }</pre>
+     */
+    public HttpRequestBuilder header(CharSequence name, Object value) {
+        requestHeadersBuilder.setObject(requireNonNull(name, "name"), requireNonNull(value, "value"));
+        return this;
+    }
+
+    /**
+     * Sets multiple headers for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .headers(HttpHeaders.of("authorization", "foo", "bar", "baz"))
+     *            .build();
+     * }</pre>
+     * @see HttpHeaders
+     */
+    public HttpRequestBuilder headers(HttpHeaders httpHeaders) {
+        requireNonNull(httpHeaders, "httpHeaders");
+        httpHeaders.forEach((BiConsumer<AsciiString, String>) requestHeadersBuilder::set);
```

Comment:
- Could accept `Iterable<? extends Entry<? extends CharSequence, String>>` instead.
- `requestHeadersBuilder.set(httpHeaders)` does the same job.